res/ext.SimpleBatchUpload.js

```diff
@@ -27,6 +27,19 @@
 	'use strict';
 
 	$( function () {
+
+		var filesLimitPerBatchConfig = mw.config.get( 'simpleBatchUploadMaxFilesPerBatch' ),
+			userGroups = mw.config.get( 'wgUserGroups' ),
+			userUploadLimit = 1000;
+
+		if ( filesLimitPerBatchConfig ) {
+			$.each( filesLimitPerBatchConfig, function ( role, limit ) {
+				if ( userGroups.indexOf( role ) !== -1 && ( userUploadLimit > limit ) ) {
```

Comment:
This loop will start with the default 1000 and then reduce it to the most restrictive role the user has. 
If the wiki has this setting to disallow uploading by anon users it will also disallow uploading by all known users, as everybody is in the '*' group:
``` php
$wgSimpleBatchUploadMaxFilesPerBatch = {
  '*': 0,
  'user': 5,
  'sysop': 500,
}
```

I think starting from `userUploadLimit = 0` and going upwards should be safe. Every user should be at least in the * user group, so the default 1000 is covered. If the wiki admin removed the '*' group from the config array (or if the config array is null) then the intention is probably to disallow uploading for anon users.